javalib/src/main/scala/java/lang/String.scala

```diff
@@ -875,16 +928,17 @@ for (cp <- 0 to Character.MAX_CODE_POINT) {
   if (!output.contains('\u0307'))
     println(cp.toHexString)
 }
-
      */
     def afterSoftDotted(i: Int): scala.Boolean = {
       val j = skipCharsWithCombiningClassOtherThanNoneOrAboveBackwards(i)
-      j > 0 && (codePointBefore(j) match {
+      def isSoftDotted(codePoint: Int): scala.Boolean = codePoint match {
         case 0x0069 | 0x006a | 0x012f | 0x0268 | 0x0456 | 0x0458 | 0x1e2d |
             0x1ecb =>
           true
         case _ => false
-      })
+      }
+      j > 0 && isSoftDotted(codePointBefore(j)) ||
+      j == 0 && isSoftDotted(codePointAt(j))
```

Comment:
Same thing here. The last condition should always be false.
```suggestion
      j > 0 && isSoftDotted(codePointBefore(j))
```